spec/controllers/authenticate_controller_spec.rb

```diff
@@ -6,6 +6,45 @@
   let(:password) { "password" }
   let(:login) { "u-#{random_hex}" }
   let(:account) { "rspec" }
+  let(:authenticator) { "authn" }
+
+  context "#login" do
+    shared_examples_for "successful authentication" do
+      it "succeeds" do
+        post :login, params
+        expect(response).to be_ok
+        expect(response.body.length).to be >= 44
+      end
+    end
+    
+    shared_examples_for "authentication denied" do
+      it "is unauthorized" do
+        post :login, account: account, authenticator: authenticator
+        expect(response.code).to eq("401")
+      end
+    end
+    
+    context "without auth" do
+      it_should_behave_like "authentication denied"
+    end
+    context "when user doesn't exist" do
```

Comment:
Mind sticking some empty lines into this block of text? At least above each `context`?